### PR TITLE
Build: check external asciidoc links

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -855,8 +855,44 @@ task prepareAsciidocBuild(type: Sync) {
 	into "$buildDir/asciidoc"
 }
 
-task('makePDF', type: org.asciidoctor.gradle.AsciidoctorTask) {
+task checkAsciidocLinks {
 	dependsOn prepareAsciidocBuild
+	inputs.dir("$buildDir/asciidoc")
```
Comment (artembilan) [inline build.gradle]: Why do you do this in `buildDir`, but not in the `src/reference/asciidoc/` ?

File: build.gradle [hunk 2]
```diff
@@ -855,8 +855,44 @@ task prepareAsciidocBuild(type: Sync) {
...
+					if (!anchors.contains(anchor)) {
+						throw new InvalidUserDataException("Anchor '" + anchor
+							+ "' not found in '" + file.name
+							+ "', if in another file, it needs to be qualified with './fileName.adoc#'")
+					}
+				}
+				else {
+					if (!hasComma) {
+						throw new InvalidUserDataException("External anchor '" + anchor
+							+ "' should have a textual description (,...)")
```
Comment (artembilan) [inline build.gradle]: Shouldn't we include a `file.name` in this exception as well?

**Review by artembilan — CHANGES_REQUESTED** [review]
A couple concerns.

File: build.gradle (hunk 1 above)
Comment (garyrussell) [reply build.gradle]: Why does it matter? The `prepareAsciidocBuild` copies them there and all other doc tasks work on that dir.

Comment (artembilan) [reply build.gradle]: Got it. Thanks!

File: build.gradle (hunk 2 above)
Comment (garyrussell) [reply build.gradle]: Good catch

**Pull request merged**